lib/manageiq_performance/middlewares/active_support_timers.rb

```diff
@@ -19,6 +19,14 @@ def active_support_timers_initialize
       def active_support_timers_start(env); end
 
       def active_support_timers_finish(env)
+        if Thread.current[:miq_perf_request_timer_data].nil? and env["HTTP_MIQ_PERF_TIMESTAMP"]
+          time = (env["MIQ_PERFORMANCE_END_TIME"] - env["HTTP_MIQ_PERF_TIMESTAMP"]) / 1000
```

Comment:
Either consistently use strings. and have the test us strings.
Or allow either, use either, and try and have tests test both cases.

Only reason for consistency is to make it easier on you.
Both work.